ktls/src/lib.rs

```diff
@@ -1,451 +1,33 @@
-pub enum KtlsCipherType {
-    AesGcm128,
-    AesGcm256,
-    Chacha20Poly1305,
-}
-
-#[derive(Debug, thiserror::Error)]
-pub enum CipherSuiteError {
-    #[error("TLS 1.2 support not built in")]
-    Tls12NotBuiltIn,
-
-    #[error("unsupported cipher suite")]
-    UnsupportedCipherSuite(SupportedCipherSuite),
-}
-
-impl TryFrom<SupportedCipherSuite> for KtlsCipherSuite {
-    type Error = CipherSuiteError;
-
-    fn try_from(#[allow(unused)] suite: SupportedCipherSuite) -> Result<Self, Self::Error> {
-        {
-            let version = match suite {
-                SupportedCipherSuite::Tls12(..) => {
-                    if !cfg!(feature = "tls12") {
-                        return Err(CipherSuiteError::Tls12NotBuiltIn);
-                    }
-                    KtlsVersion::TLS12
-                }
-                SupportedCipherSuite::Tls13(..) => KtlsVersion::TLS13,
-            };
-
-            let family = {
-                if suite == cipher_suite::TLS13_AES_128_GCM_SHA256 {
-                    KtlsCipherType::AesGcm128
-                } else if suite == cipher_suite::TLS13_AES_256_GCM_SHA384 {
-                    KtlsCipherType::AesGcm256
-                } else if suite == cipher_suite::TLS13_CHACHA20_POLY1305_SHA256 {
-                    KtlsCipherType::Chacha20Poly1305
-                } else if suite == cipher_suite::TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 {
-                    KtlsCipherType::AesGcm128
-                } else if suite == cipher_suite::TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 {
-                    KtlsCipherType::AesGcm256
-                } else if suite == cipher_suite::TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 {
-                    KtlsCipherType::Chacha20Poly1305
-                } else {
-                    return Err(CipherSuiteError::UnsupportedCipherSuite(suite));
-                }
-            };
-
-            Ok(Self {
-                typ: family,
-                version,
-            })
-        }
-    }
-}
-
-impl KtlsCipherSuite {
-    pub fn as_supported_cipher_suite(&self) -> SupportedCipherSuite {
-        match self.version {
-            KtlsVersion::TLS12 => match self.typ {
-                KtlsCipherType::AesGcm128 => cipher_suite::TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
-                KtlsCipherType::AesGcm256 => cipher_suite::TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
-                KtlsCipherType::Chacha20Poly1305 => {
-                    cipher_suite::TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
-                }
-            },
-            KtlsVersion::TLS13 => match self.typ {
-                KtlsCipherType::AesGcm128 => cipher_suite::TLS13_AES_128_GCM_SHA256,
-                KtlsCipherType::AesGcm256 => cipher_suite::TLS13_AES_256_GCM_SHA384,
-                KtlsCipherType::Chacha20Poly1305 => cipher_suite::TLS13_CHACHA20_POLY1305_SHA256,
-            },
-        }
-    }
-}
+pub mod log;
+mod protocol;
+pub mod setup;
+pub mod stream;
+pub mod utils;
```

Comment:
Instead of this being `pub` we should just export `CompatibleCipherSuites` in the crate root.